packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,276 @@
+  primaryAction,
+  offsetTop = 130,
+  offsetBottom = 24,
+  thumbnail,
+  zIndex,
+}: Props): ReactNode {
+  const { name: colorSchemeName } = useColorScheme();
+  const isDarkMode = colorSchemeName === 'darkMode';
+
+  const isMobileWidth = useDeviceType() === 'mobile';
+
+  let messageTextElement: Element<'span'> | string;
+
+  if (typeof message === 'string') {
+    messageTextElement = message;
+  }
+  if (typeof title === 'string') {
+    messageTextElement = title;
+  }
+
+  // If `message` is a Text component, we need to override any text colors within to ensure they all match
+  const messageIsTextNode =
+    typeof message !== 'string' &&
+    Children.only<Element<typeof Text>>(message).type.displayName === 'Text';
+
+  if (messageIsTextNode) {
+    const textColorOverrideStyles = isDarkMode
+      ? styles.textColorOverrideDark
+      : styles.textColorOverrideLight;
+
+    messageTextElement = <span className={textColorOverrideStyles}>{message}</span>;
+  }
+
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('BannerOverlay');
+
+  const { lightModeBackground, darkModeBackground, textColor } = DEFAULT_COLORS;
+
+  const dismissButtonComponent = dismissButton && (
+    <Flex.Item flex="none" alignSelf={isMobileWidth ? 'end' : 'center'}>
+      <IconButton
+        accessibilityLabel={
+          dismissButton.accessibilityLabel ?? accessibilityDismissButtonLabelDefault
+        }
+        icon="cancel"
+        iconColor="darkGray"
+        onClick={dismissButton.onDismiss}
+        size="xs"
+      />
+    </Flex.Item>
+  );
+  const messageComponent = (
+    <Flex.Item flex="grow">
+      <Text weight="bold">{title}</Text>
+      <BannerOverlayMessage
+        text={messageIsTextNode ? undefined : messageTextElement}
+        textElement={messageIsTextNode ? messageTextElement : undefined}
+        helperLink={helperLink}
+        textColor={textColor}
+      />
+    </Flex.Item>
+  );
+  const CTAComponent = primaryAction && (
+    <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
+      {primaryAction?.accessibilityLabel && primaryAction?.label
+        ? (primaryAction.role === 'link' && (
+            <PrimaryAction
+              accessibilityLabel={primaryAction.accessibilityLabel}
+              href={primaryAction.href}
+              label={primaryAction.label}
+              onClick={primaryAction.onClick}
+              rel={primaryAction?.rel}
+              role="link"
+              size="sm"
+              target={primaryAction?.target}
+            />
+          ),
+          primaryAction.role !== 'link' && (
+            <PrimaryAction
+              accessibilityLabel={primaryAction.accessibilityLabel}
+              label={primaryAction.label}
+              onClick={primaryAction.onClick}
+              role="button"
+              size="sm"
+            />
+          ))
+        : null}
+    </Flex.Item>
+  );
+  return (
+    <Layer>
```

Comment:
This component is just not ging to be accessible... but I don't know how to it at the moment. It's goig to be at the bottom of DOM tree. And it's rendered since the page loads... so it's kind of unreachable.

Can we make some research on how to make floating Banners keyboard accessible?

Reply:
I'll try researching a bit, but I assume this is more of an issue with Layer. Any time it is used, the children in the tree are not accessible. We should consider a solution for this considering the issue with Layer itself later, though.

Reply:
Got rid of the Layer component entirely, opting instead for a css-based approach.